src/test/java/wolox/training/controllersTest/userRepositoryTest.java

```diff
@@ -0,0 +1,228 @@
+package wolox.training.controllersTest;
+
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.controllers.UserController;
+import wolox.training.entitiesTest.EntitiesTest;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(controllers = UserController.class)
+public class userRepositoryTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    @MockBean
+    private PasswordEncoder passwordEncoder;
+
+    private static User userTest;
+    private static User secondUserTest;
+    private static Book bookTest;
+    private static List<User> userTests;
+    private static List<Book> bookTests;
+    private static final String PATH = "/api/users";
+
+    @BeforeAll
```

Comment:
this annotation should be @beforeeach. This way the objects are initialized and eliminated every time a test is executed

Reply:
I changed the annotation